enhancements/bundle-validation.md

```diff
@@ -0,0 +1,235 @@
+    - E48: duplicate package manifest channel name
+    - E49: package channel.currentCSV not defined
+    - E50: channel.currentCSV is not included in list of csvs
+    - E51: The packageName in bundle does not match repository name
+    - E52: duplicate CSV in bundle set
+    - E53: error getting spec.replaces from bundle CSV
+    - E54: spec.replaces is referenced by more than one CSV
+    - E55: spec.replaces field cannot match its own metadata.name
+    - E56: spec.replaces CSV is not present in manifests
+    - E57: spec.skips cannot match spec.replaces
+    - E58: metadata.annotations.skipPackageAnnotationKey range contains the current CSV's version
+    - E59: metadata.annotations.skipPackageAnnotationKey is an invalid semantic version range
+    - E60: spec.replaces references a parent in CSV replace chain which causes repalce cycle
+    - E61: spec.replaces does not map to any CSV in bundles
+    - E62: currentCSV for channel name not found in bundle
+
+- Validating cluster service versions for operatorhub.io UI. **\***
+    - W5: CRD does not have an example CR in alm-examples
+    - W6: every owned CRD should have alm-examples
+    - E63: csv.spec.provider should be a singleton list
+    - E64: csv.spec.provider element should contain name as a single field
+    - E65: csv.spec.maintainers element should contain both name and email
+    - E66: csv.spec.maintainers element email is not valid
+    - E67: csv.spec.maintainers must be a list of name & email pairs
+    - E68: csv.spec.links element should contain both name and url
+    - E69: csv.spec.links element url is not valid
+    - E70: csv.spec.links must be a list of name & url pairs
+    - E71: spec.version is not a valid semver
+    - E72: metadata.annotations.capabilities is not a valid capabilities level if it is not ["Basic Install", "Seamless Upgrades", "Full Lifecycle", "Deep Insights", "Auto Pilot"]
+    - E73: spec.icon.mediatype %s is not a valid mediatype if it is not [\"image/gif\", \"image/jpeg\", \"image/png\", \"image/svg+xml\"]
+    - E74: spec.icon can only contain two fields: \"base64data\" and \"mediatype\"
+    - E75: spec.icon should be a singleton list
+    - E76: category is not valid if it is not [
+                "AI/Machine Learning",
+                "Application Runtime",
+                "Big Data",
+                "Cloud Provider",
+                "Developer Tools",
+                "Database",
+                "Integration & Delivery",
+                "Logging & Tracing",
+                "Monitoring",
+                "Networking",
+                "OpenShift Optional",
+                "Security",
+                "Storage",
+                "Streaming & Messaging"
+            ]
+
+**\*** Validation may be subjected to relocation or included as part of extended package.
+
+#### Public validation output struct
+
+```go
+type alert struct {
+  source    string
+  code      string
+  message   string
+  severity  string
```

Comment:
I assume we would actually want a type for this. Can we follow up with types for severity and potentially for code as well?